src/core/border_router/routing_manager.cpp

```diff
@@ -1174,6 +1231,14 @@ void RoutingManager::UpdateRouterAdvertHeader(const Ip6::Nd::RouterAdvertMessage
 
     Ip6::Nd::RouterAdvertMessage::Header oldHeader;
 
+    if (aRouterAdvertMessage != nullptr)
+    {
+        // We skip and do not update RA header if the received RA message
+        // was not prepared and sent by `RoutingManager` itself.
+
+        VerifyOrExit(!IsReceivdRouterAdvertFromManager(*aRouterAdvertMessage));
```

Comment:
I guess IsReceivdRouterAdvertFromManager might return true for  RAs from another BR of the same Thread network?